core-tests/shared/src/test/scala/zio/stm/ZSTMSpec.scala

```diff
@@ -18,6 +17,37 @@ object ZSTMSpec extends ZIOBaseSpec {
       testM("`STM.failed` to make a failed computation and check the value") {
         assertM(STM.failNow("Bye bye World").commit.run)(fails(equalTo("Bye bye World")))
       },
+      suite("`absolve` to convert ")(
+        testM("A successful Right computation into the success channel") {
+          assertM(STM.succeedNow(Right(42)).absolve.commit)(equalTo(42))
+        },
+        testM("A successful Left computation into the error channel") {
+          assertM(STM.succeedNow(Left("oh no!")).absolve.commit.run)(fails(equalTo("oh no!")))
+        }
+      ),
+      suite("`bimap` when")(
+        testM("having a success value") {
+          import zio.CanFail.canFail
+          assertM(STM.succeedNow(1).bimap(_ => -1, s => s"$s as string").commit)(equalTo("1 as string"))
+        },
+        testM("having a fail value") {
+          assertM(STM.failNow(-1).bimap(s => s"$s as string", _ => 0).commit.run)(fails(equalTo("-1 as string")))
+        }
+      ),
+      testM("`doWhile` to run effect while it satisfies predicate") {
```

Comment:
You can do this one and the one below in a single transaction instead of committing after each step. Generally better to stay within the transactional context if possible.